patches/nginx-1.11.2-resolve-hosts-file.patch

```diff
@@ -0,0 +1,413 @@
++                              "instead of %z", n, size);
++                goto fail;
++            }
++
++            b.pos = b.start + len;
++            b.last = b.pos + n;
++            start = b.start;
++        }
++
++        ch = *b.pos;
++
++        switch (state) {
++
++        case scan_line:
++            if (ch == ' ') {
++                break;
++            }
++
++            if (ch == '#') {
++                state = scan_skipline;
++                break;
++            }
++
++            if (ch != LF && ch != CR) {
++                start = b.pos;
++                state = scan_addr;
++            }
++
++            break;
++
++        case scan_skipline:
++            if (ch == LF || ch == CR) {
++                state = scan_line;
++            }
++
++            break;
++
++        case scan_addr:
++            if (ch == LF || ch == CR) {
++                state = scan_line;
++                break;
++            }
++
++            if (ch == ' ' || ch == '\t') {
++                if (s.data) {
++                    ngx_free(s.data);
++                }
++
++                s.len = b.pos - start;
++
++                s.data = ngx_alloc(s.len, cf->log);
++                if (s.data == NULL) {
++                    goto fail;
++                }
++
++                ngx_memcpy(s.data, start, s.len);
++
++                state = scan_hosts;
++            }
++
++            break;
++
++        case scan_hosts:
++            if (ch == LF || ch == CR) {
++                state = scan_line;
++                break;
++            }
++
++            if (ch == ' ' || ch == '\t') {
++                break;
++            }
++
++            start = b.pos;
++            state = scan_name;
++            break;
++
++        case scan_name:
++            if (ch == ' ' || ch == '\t' || ch == LF || ch == CR) {
++                rn = ngx_calloc(sizeof(ngx_resolver_node_t), cf->log);
++                if (rn == NULL) {
++                    goto fail;
++                }
++
++                rn->nlen = b.pos - start;
++
++                rn->name = ngx_alloc(rn->nlen, cf->log);
++                if (rn->name == NULL) {
++                    goto fail;
++                }
++
++                ngx_memcpy(rn->name, start, rn->nlen);
++
++                rn->ttl = NGX_MAX_UINT32_VALUE;
++                rn->valid = NGX_MAX_UINT32_VALUE;
++                rn->expire = NGX_MAX_UINT32_VALUE;
++                rn->node.key = ngx_crc32_short(rn->name, rn->nlen);
++
++                if (ngx_strlchr(s.data,
++                                s.data + s.len, ':') != NULL)
++                {
++
++#if (NGX_HAVE_INET6)
++                    if (!r->ipv6
++                        || ngx_inet6_addr(s.data, s.len,
++                                          rn->u6.addr6.s6_addr) != NGX_OK)
++                    {
++                        ngx_resolver_free_node(r, rn);
++                        state = scan_skipline;
++                        break;
```

Comment:
Here, we are missing such a `break` when the parsed value is an ipv6 and when Nginx is built *without* ipv6 support. Will update.